### PR TITLE
Feature/detail answer student

### PR DESCRIPTION
**Pull request by Francois389**
#169 

# Result

The teacher can see all information about a student submission from the dashboard
Détail-etudiant

If the student didn't answer, we didn't fetch the resource.
Détail-etudiant-Sans-réponse

## Technical point 

The content of the modal is loaded only one time. Before fetching, we check if the container is empty.

### CODE REVIEW

File: src/main/kotlin/org/elaastic/questions/assignment/sequence/SequenceService.kt [hunk 1]
```diff
@@ -81,6 +82,14 @@ class SequenceService(
         } ?: throw EntityNotFoundException("There is no sequence for id \"$id\"")
     }
 
+    // TODO: delete this method!
+    @Deprecated("Will be deleted",
```
Comment (FranckSilvestre) [inline src/main/kotlin/org/elaastic/questions/assignment/sequence/SequenceService.kt:86]: Humm, is it normal to see a method to be deleted, added?
A rebase issue?

File: src/main/kotlin/org/elaastic/questions/assignment/sequence/SequenceService.kt [hunk 2]
```diff
@@ -47,6 +47,7 @@ import org.springframework.stereotype.Service
 import java.util.UUID
 import javax.persistence.EntityNotFoundException
 import javax.transaction.Transactional
+import kotlin.jvm.Throws
```
Comment (FranckSilvestre) [inline src/main/kotlin/org/elaastic/questions/assignment/sequence/SequenceService.kt:50]: What is this import?

File: src/main/kotlin/org/elaastic/questions/player/PlayerController.kt [hunk 3]
```diff
@@ -709,7 +708,7 @@ class PlayerController(
     ): String {
         val user: User = authentication.principal as User
         val sequence = sequenceService.get(id, true)
-
+        // TODO test if the learner can report the evaluation
```
Comment (FranckSilvestre) [inline src/main/kotlin/org/elaastic/questions/player/PlayerController.kt:711]: Why this TODO?

File: src/main/kotlin/org/elaastic/questions/player/PlayerController.kt [hunk 4]
```diff
@@ -723,4 +722,60 @@ class PlayerController(
...
+        model: Model,
+        @PathVariable sequenceId: Long,
+        @PathVariable userId: Long
+    ): String {
+        val user: User = authentication.principal as User
+
+        val sequence = sequenceService.get(sequenceId, true)
+        val learner = userService.findById(userId)
+
+        // TODO change requireAccess to requireAccessThrowDenied when fix/249-cant-see-chatgpt-feedback-with-likert will be merged
```
Comment (FranckSilvestre) [inline src/main/kotlin/org/elaastic/questions/player/PlayerController.kt:746]: Is-it possible to fix this TODO now ?

**Review by FranckSilvestre — CHANGES_REQUESTED** [review]
Hi
Just some points to modify, probably small issues coming from one of the last rebase. 
The CI is broken too. We need to fix it.

File: src/main/kotlin/org/elaastic/questions/assignment/sequence/SequenceService.kt (hunk 1 above)
Comment (Francois389) [reply src/main/kotlin/org/elaastic/questions/assignment/sequence/SequenceService.kt:86]: Yes, it's a rebase issue.

Fix in 8452777041ef5c91bfbe746ac4612636f48212df

File: src/main/kotlin/org/elaastic/questions/player/PlayerController.kt (hunk 4 above)
Comment (Francois389) [reply src/main/kotlin/org/elaastic/questions/player/PlayerController.kt:746]: Fix in e1d442f5532d0f6068acc657326bc661b774e12e

File: src/main/kotlin/org/elaastic/questions/player/PlayerController.kt (hunk 3 above)
Comment (Francois389) [reply src/main/kotlin/org/elaastic/questions/player/PlayerController.kt:711]: It's a security issue. With small impact but still.

If somebody know the URL, the sequence ID and the evaluation ID, he can report the evaluation.
With the current implementation only the learner of the response evaluate by ChatGPT, can access the user interface and so report it.

To fix this issue : 
- retrieve the response evaluate by ChatGPT
- check if the `response.learner` is the user given by `authentication.principal as User`

Comment (Francois389) [reply src/main/kotlin/org/elaastic/questions/player/PlayerController.kt:711]: Should I fix it ?

Comment (Francois389) [reply src/main/kotlin/org/elaastic/questions/player/PlayerController.kt]: Fix in f64bbc694593592b8d7ad11d40e76f72714c8cf3

**Review by FranckSilvestre — APPROVED** [review]

**Pull request merged**